### DISCUSSION TITLE
How to make LazyVim to undo changes made in the current session only without affecting changes saved in the previous sessions?

### DISCUSSION THREAD

**Discussion by hopezh** [LazyVim/LazyVim#789] [Q&A] [answered]
I added two test lines to the end of this file and save the changes and quite. 
![image](https://github.com/LazyVim/LazyVim/assets/23435810/d2b808f7-ad52-499c-a003-cd14643f1545)

Once I open this file, I can use `u` to perform `undo` to remove the test lines, i.e. the `undo` cmd in LazyVim can affect all the changes made in the previous session. This could lead to unwanted results if not pressing `u` carefully... 
![image](https://github.com/LazyVim/LazyVim/assets/23435810/c4d64d5b-f153-42a4-b11a-28c769585e5d)

May I ask which setting in LazyVim is related to this "no-upper-cap undo" behavior that I can switch off? 

Thanks.

**Comment by folke** [comment]
As far as I know there is no such setting in Neovim. undo in Neovim is also unrelated to sessions

**Reply by hopezh** [reply]
Thanks. Then, this behavior in my LazyVim config is quite strange, because I don't have such an issue in other configs, either preset, or assembled by myself... Can you confirm you get the same undo behavior in your config?

**Reply by folke** [reply]
You probably mean you don't have this in vim? NEovim  has shada and undo files where these things are tracked. Again, all unrelated to sessions or LazyVim.

I also don't understand why you would not want to be able to undo changes from previous session. Makes no sense to me tbh

**Reply by hopezh** [reply]
The changes made and saved before I open the current working file could be many lines away from the lines I'm editing now. So, I may not even be aware that they have been reverted back when I press "u" a few more times accidentally while thinking I'm only undo-ing some changes in the current position. 

My understanding is that vim/neovim will show "**_Already at oldest change_**" when continuously pressing "u" hit the bottom, or top, i.e. the file is at the state when it's opened, and there's no more change to be "undo". Am I understanding it correctly? 

So, the issue here is: LazyVim doesn't show "Already at oldest change" when the file has returned to the state when opened in the current session, and more "undo" will revert changes made before the current file is opened. I just want to turn off this function.

**Reply by folke** [reply, marked as answer]
You can disable undo files to get the behavior you want.

```lua
-- ~/.config/nvim/lua/config/options.lua
vim.opt.undofile = false
```
